test/network/web_socket_peer_test.dart

```diff
@@ -51,23 +81,74 @@ void main() {
   setUp(() async {
     devices = [
       await Device.create(name: 'device1', port: 58240),
-      await Device.create(name: 'device2', port: 58241),
+      await Device.create(name: 'device2', port: 58242),
     ];
     await Future.wait(devices.map(
       (device) => device.peerService.startServer(),
     ));
   });
 
+  group('Introduction message', () {
+    test('should add new peer info on the receiving device', () async {
+      final peerInfoDevice0 = await devices[0].generatePeerInfo();
+      final peerInfoDevice1 = await devices[1].generatePeerInfo();
+      final currentPeerInfo =
+          await devices[1].peerInfoService.getByID(peerInfoDevice0.id!);
+      expect(currentPeerInfo, equals(null));
+
+      await devices[0].peerService.sendIntroductionMessageToPeer(
+            peerInfoDevice1,
+            location: peerInfoDevice1.locations.first,
+          );
+
+      final newPeerInfo =
+          await devices[1].peerInfoService.getByID(peerInfoDevice0.id!);
+      expect(newPeerInfo, isNot(equals(null)));
```

Comment:
`isNot(null)` should suffice, I don't think the `equals` is needed (also in later tests)